lua/nvim-treesitter/textobjects/shared.lua

```diff
@@ -35,11 +38,28 @@ function M.textobject_at_point(query_string)
     end
   end
 
+  if trynext then
+    -- Sort by order of appearance
+    table.sort(matches, function (a, b)
+      local _, _, start_a = a.node:start()
+      local _, _, start_b = b.node:start()
+      return start_a < start_b
+    end)
+  end
```

Comment:
I planned to rewrite textobject at point in terms of `select_best_match` (= it would be the same call as in `goto_next`). This should de-duplicate the code.

Reply:
Not sure which function you refer to

Reply:
All other functions for textobjects use:
`local next_node = queries.find_best_match(bufnr, query_string, 'textobjects', filter_function, scoring_function)`
`find_best_match` is essentially the same code as `textobject_at_point`. It just wasn't replaced because it doesn't support `@function.outer.start` yet.